packages/shared/lib/core/wallet/actions/activities/generateAndStoreActivitiesForAccount.ts

```diff
@@ -0,0 +1,23 @@
+import { IAccountState } from '@core/account'
+import { setAsyncActivitiesToClaimed } from '../setAsyncActivitiesToClaimed'
+import { preprocessAccountTransactions } from './preprocessAccountTransactions'
+import { preprocessAccountOutputs } from './preprocessAccountOutputs'
+import { linkActivityAndClaimingTransaction } from './linkActivityAndClaimingTransaction'
+import { hideActivitiesForFoundries } from './hideActivitiesForFoundries'
+import { generateActivitiesFromProcessedTransactions } from './generateActivitiesFromProcessedTransactions'
+import { setAccountActivitiesInAllAccountActivities } from '@core/wallet/stores'
+
+export async function generateAndStoreActivitiesForAccount(account: IAccountState): Promise<void> {
+    // Step 1: process account transactions and outputs into processed transactions
+    const processedTransactions = [...preprocessAccountTransactions(account), ...preprocessAccountOutputs(account)]
+
+    // Step 2: generate activities from processed transactions
+    const activities = generateActivitiesFromProcessedTransactions(processedTransactions, account)
+
+    // Step 3: set account activities with generated activities
+    setAccountActivitiesInAllAccountActivities(account.id, activities)
+
+    hideActivitiesForFoundries(account)
```

Comment:
Now we group outputs together do we need to hide activities for foundries because only one would have been created? And in the future we might want to create more than one activity out of the processedTransaction, such that an alias was also created.